app/src/views/features/sessions/SessionViewer/DraftSessionViewer.tsx

```diff
@@ -96,46 +146,19 @@ const DraftSessionViewer: React.FC = () => {
 
   useEffect(() => {
     trackDraftSessionViewed();
-
     setIsLoading(true);
 
     if (tabId === "imported") {
       setIsLoading(false);
     } else if (tabId === "mock") {
-      // TODO: remove mock flow
-      dispatch(
-        sessionRecordingActions.setSessionRecordingMetadata({
-          sessionAttributes: mockSession.attributes,
-          name: "Mock Session Recording",
-        })
-      );
-      dispatch(sessionRecordingActions.setEvents(mockSession.events));
+      populateSessionData(mockSession.attributes, mockSession.events, tabId);
       setIsLoading(false);
+    } else if (queryParams.has("savedDraft")) {
+      handleGetSavedDraftReplay();
     } else {
-      getTabSession(parseInt(tabId)).then((payload: unknown) => {
-        if (typeof payload === "string") {
-          setLoadingError(payload);
-        } else {
-          const tabSession = payload as RQSession;
-
-          if (tabSession.events.rrweb?.length < 2) {
-            setLoadingError("RRWeb events not captured");
-          } else {
-            dispatch(
-              sessionRecordingActions.setSessionRecordingMetadata({
-                sessionAttributes: tabSession.attributes,
-                name: generateDraftSessionTitle(tabSession.attributes?.url),
-              })
-            );
-
-            filterOutLargeNetworkResponses(tabSession.events);
-            dispatch(sessionRecordingActions.setEvents(tabSession.events));
-          }
-        }
-        setIsLoading(false);
-      });
+      handleGetNewDraftReplay();
     }
-  }, [dispatch, tabId, user?.details?.profile?.email, generateDraftSessionTitle]);
+  }, [dispatch, tabId, queryParams, handleGetSavedDraftReplay, handleGetNewDraftReplay, populateSessionData]);
 
   const confirmDiscard = () => {
```

Comment:
<img width="1083" alt="Screenshot 2023-08-13 at 7 38 27 PM" src="https://github.com/requestly/requestly/assets/56021937/b52c297e-57a4-4306-ac14-3d27866257db">
I see 2 confirmation modals on clicking discard. I don't think the custom modal is needed. `window.confirm` is  sufficient as it is triggered on route change.